File: src/WebApp/CommerceCore.Web.AdminSite/src/shared/constants/product.ts
type ProductLabelKey =
  | "id"
  | "name"
  | "description"
  | "price"
  | "createdDate"
  | "updatedDate"
  | "category"
  | "files";

export const productLabels: Record<ProductLabelKey, string> = {
  id: "Id",
  name: "Name",
  description: "Description",
  price: "Price",
  createdDate: "Created Date",
  updatedDate: "Updated Date",
  category: "Category",
  files: "Files",
};

File: src/ResourceServer/CommerceCore.Domain/Entities/Product.cs
namespace CommerceCore.Domain.Entities;

public class Product
{
    public Guid Id { get; init; }
    public required string Name { get; init; }
    public required string Description { get; init; }
    public required decimal Price { get; init; }
    public DateTime CreatedDate { get; init; }
    public DateTime UpdatedDate { get; init; }

    public virtual Category Category { get; init; } = null!;
    public virtual ICollection<ProductVariant> Variants { get; init; } = [];
}
